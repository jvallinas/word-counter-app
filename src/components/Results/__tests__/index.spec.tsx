import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Result from "..";
import { EXPECTED_RESULT_ALPHABETICAL_ASCENDING } from "../../../hooks/__tests__/index.spec";

configure({ adapter: new Adapter() });

describe("Results component", () => {
  describe("when word entries are passed", () => {
    it("should render properly", () => {
      const component = shallow(
        <Result wordEntries={EXPECTED_RESULT_ALPHABETICAL_ASCENDING}></Result>
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe("when no word entries are passed", () => {
    it("should render properly", () => {
      const component = shallow(<Result wordEntries={[]} />);

      expect(component).toMatchSnapshot();
    });
  });
});
