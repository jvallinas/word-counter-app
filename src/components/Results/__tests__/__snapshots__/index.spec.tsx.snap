// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component when no word entries are passed should render properly 1`] = `
<Table>
  <HeaderRow>
    <CellWordCount>
      Word
    </CellWordCount>
    <styled.div>
      Occurrences
    </styled.div>
  </HeaderRow>
  <NoResultsRow>
    No entries available
  </NoResultsRow>
</Table>
`;

exports[`Results component when word entries are passed should render properly 1`] = `
<Table>
  <HeaderRow>
    <CellWordCount>
      Word
    </CellWordCount>
    <styled.div>
      Occurrences
    </styled.div>
  </HeaderRow>
  <Row
    key="A"
  >
    <CellWordCount>
      A
    </CellWordCount>
    <styled.div>
      1
    </styled.div>
  </Row>
  <Row
    key="hook"
  >
    <CellWordCount>
      hook
    </CellWordCount>
    <styled.div>
      1
    </styled.div>
  </Row>
  <Row
    key="input"
  >
    <CellWordCount>
      input
    </CellWordCount>
    <styled.div>
      1
    </styled.div>
  </Row>
  <Row
    key="mock"
  >
    <CellWordCount>
      mock
    </CellWordCount>
    <styled.div>
      1
    </styled.div>
  </Row>
  <Row
    key="sentence"
  >
    <CellWordCount>
      sentence
    </CellWordCount>
    <styled.div>
      1
    </styled.div>
  </Row>
  <Row
    key="short"
  >
    <CellWordCount>
      short
    </CellWordCount>
    <styled.div>
      1
    </styled.div>
  </Row>
  <Row
    key="some"
  >
    <CellWordCount>
      some
    </CellWordCount>
    <styled.div>
      1
    </styled.div>
  </Row>
  <Row
    key="test"
  >
    <CellWordCount>
      test
    </CellWordCount>
    <styled.div>
      1
    </styled.div>
  </Row>
  <Row
    key="this"
  >
    <CellWordCount>
      this
    </CellWordCount>
    <styled.div>
      1
    </styled.div>
  </Row>
  <Row
    key="to"
  >
    <CellWordCount>
      to
    </CellWordCount>
    <styled.div>
      2
    </styled.div>
  </Row>
</Table>
`;
